### PR TITLE
cmake: function for application creation

### PR DESCRIPTION
**Pull request by mshabunin**
to be merged after #11734

### This pullrequest changes

Added single cmake function for application (<opencv>/apps) creation.

### CODE REVIEW

File: apps/createsamples/CMakeLists.txt [hunk 1]
```diff
@@ -1,38 +1,4 @@
...
-endif()
-
-project(createsamples)
-set(the_target opencv_createsamples)
-
-ocv_target_include_directories(${the_target} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}" "${OpenCV_SOURCE_DIR}/include/opencv")
-ocv_target_include_modules_recurse(${the_target} ${OPENCV_CREATESAMPLES_DEPS})
-
 file(GLOB SRCS *.cpp)
-file(GLOB HDRS *.h*)
```
Comment (alalek) [inline apps/createsamples/CMakeLists.txt]: BTW, non-included header files are not showed in MSVS solutions.

File: apps/version/CMakeLists.txt [hunk 2]
```diff
@@ -1,32 +1,5 @@
...
-if(INSTALL_CREATE_DISTRIB)
-  if(BUILD_SHARED_LIBS)
-    install(TARGETS ${the_target} RUNTIME DESTINATION ${OPENCV_BIN_INSTALL_PATH} CONFIGURATIONS Release COMPONENT libs)
-  endif()
-else()
-  install(TARGETS ${the_target} RUNTIME DESTINATION ${OPENCV_BIN_INSTALL_PATH} COMPONENT libs)
+ocv_add_application(opencv_version MODULES opencv_core SRCS opencv_version.cpp)
+if(WIN32)
+  ocv_add_application(opencv_version_win32 MODULES opencv_core SRCS opencv_version.cpp)
+  target_compile_definitions(opencv_version_win32 PRIVATE "OPENCV_WIN32_API=1")
```
Comment (alalek) [inline apps/version/CMakeLists.txt]:
```
if(TARGET opencv_version_win32)
  target_compile_definitions(...)
endif()
```

for case of 'return' from function (missing dependencies).

Actually it is not required for this case, but may be useful for others.

**Comment by vpisarev** [comment]
:+1:

**Comment by vpisarev** [comment]
let's put it in :+1:

**Pull request merged**